packages/dev-console/src/types.ts

```diff
@@ -1,60 +1,42 @@
-type DeepPartial<T> = {
-  [P in keyof T]?: DeepPartial<T[P]>;
-};
+/* eslint-disable @typescript-eslint/no-namespace, @shopify/strict-component-boundaries */
+import './ExtensionServerClient/types';
 
-export interface DevServerUpdateCall {
-  event: 'update';
-  data: {
-    extensions?: ({uuid: string} & DeepPartial<ExtensionPayload>)[];
-    app?: DeepPartial<App>;
-  };
-}
+declare global {
+  namespace ExtensionServer {
+    interface InboundEvents {
+      connected: {extensions?: ExtensionPayload[]; app?: App};
+      update: {extensions?: ExtensionPayload[]; app?: App};
+      refresh: {uuid: string}[];
+      focus: {uuid: string}[];
+      unfocus: void;
+    }
 
-export type DevServerCall =
-  | {
-      event: 'update';
-      data: {
+    interface OutboundPersistEvents {
+      update: {
         extensions?: ({uuid: string} & DeepPartial<ExtensionPayload>)[];
         app?: DeepPartial<App>;
       };
     }
-  | {
-      event: 'dispatch';
-      data: ConsoleAction;
-    };
 
-export type DevServerResponse =
-  | {
-      event: 'update' | 'connected';
-      data: {extensions: ExtensionPayload[]; app?: App; store: string};
+    interface OutboundDispatchEvents {
+      refresh: {uuid: string}[];
+      focus: {uuid: string}[];
+      unfocus: void;
     }
-  | {
-      event: 'dispatch';
-      data: ConsoleAction;
-    };
-
-export type ConsoleAction =
-  | Action<'refresh', {uuid: string}[]>
-  | Action<'focus', {uuid: string}[]>
-  | BaseAction<'unfocus'>;
-
-export interface BaseAction<T extends string> {
-  type: T;
-}
-// @shopify/eslint-plugins doesn't allow any generic other than T
-// If this becomes a problem, consider modifying this rule
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export interface Action<T extends string = string, P = any> {
-  type: T;
-  payload: P;
+  }
```

Comment:
I like the naming Outbound and Inbound events, but is the `Record<>` structure more readable than `ConsoleActions = ` unions? It feels less intuitive here.